the-incredible-bulk/src/main/java/gov/va/api/health/bulkfhir/service/controller/bulkstatus/BulkStatusController.java

```diff
@@ -0,0 +1,156 @@
+                    .type("Patient")
+                    .url(
+                        bulkFileBaseUrl
+                            + "/"
+                            + file.publicationId()
+                            + "/"
+                            + file.fileName()
+                            + ".ndjson")
+                    .build())
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Decode the publication request string into a ResourceIdentity object.
+   *
+   * @param publicationRequestString The string to decode
+   * @return The decoded ResourceIdentity or null if one could not be found.
+   */
+  private ResourceIdentity decodePublicationRequest(String publicationRequestString) {
+    try {
+      List<ResourceIdentity> foundIdentities = identityService.lookup(publicationRequestString);
+      if (foundIdentities == null || foundIdentities.isEmpty()) {
+        return null;
+      }
+      return foundIdentities.get(0);
+    } catch (BadId | LookupFailed e) {
+      log.error("Failed to decode publication request string {}", publicationRequestString, e);
+      return null;
+    }
+  }
+
+  /**
+   * Build and return the bulk status response.
+   *
+   * @param publicationRequestString The encoded publication request containing the original
+   *     kick-off request and the publication id to retrieve
+   * @return The publication file status with a link to all of the relevant files
+   */
+  @GetMapping(path = "{id}")
+  public ResponseEntity<PublicationFileStatusResponse> getBulkStatus(
+      @RequestHeader("Accept") String acceptHeader,
+      @PathVariable("id") String publicationRequestString) {
+    if (!VALID_ACCEPT_HEADER_VALUES.contains(acceptHeader)) {
+      log.info("Invalid Accept header received for bulk status request");
+      return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+    }
+
+    ResourceIdentity publicationRequest = decodePublicationRequest(publicationRequestString);
+    if (publicationRequest == null) {
+      return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+    List<StatusEntity> fileStatuses =
+        repository.findByPublicationId(publicationRequest.identifier());
+    if (fileStatuses == null || fileStatuses.isEmpty()) {
+      /*
+       * No statuses were found for the given publication id
+       */
+      log.info("No publication data found for publication {}", publicationRequest.identifier());
+      return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+    List<PublicationFileStatusResponse.FileLocation> outputFiles =
+        createFileOutputList(fileStatuses);
+    /*
+     * Pull out the first status entity to use for the boiler plate information of the response
+     */
+    StatusEntity firstStatusEntity = fileStatuses.get(0);
+    return ResponseEntity.ok(
+        PublicationFileStatusResponse.builder()
+            .requiresAccessToken(true)
+            .request(bulkBaseUrl + publicationRequest.resource())
```

Comment:
Looks like this should include `/$export` as well.

Reply:
The `publicationRequest.resource()` has the full path beyond the base URL for the request that was made, so that should be included already as part of the request that's passed.